tableaudocumentapi/datasource.py

```diff
@@ -116,6 +127,20 @@ def from_file(cls, filename):
         dsxml = xml_open(filename, cls.__name__.lower()).getroot()
         return cls(dsxml, filename)
 
+    @classmethod
+    def from_scratch(cls, caption, connections):
+        root = ET.Element('datasource', caption=caption, version='10.0')
```

Comment:
Yes, I was thinking about some sort of DSBuilder or something, the direct XML feels gross.

It would also allow us to specify different versions (at least 9.3 vs 10)